### PR TITLE
Fix InitialRegistrationTime in case of reregistration

### PR DESCRIPTION
**Pull request by glazychev-art**
## Description
We should not change the `InitialRegistrationTime` field if we have already registered the NSE before.

Added registry `metadata` chain element

## Issue link
https://github.com/networkservicemesh/deployments-k8s/issues/9063

## How Has This Been Tested?

- [x] Added unit testing to cover
- [ ] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [ ] New functionality
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/registry/common/setregistrationtime/nse_server.go [hunk 1]
```diff
@@ -19,15 +21,19 @@ package setregistrationtime
...
+	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
 	"github.com/networkservicemesh/sdk/pkg/tools/clock"
 )
 
-type setregtimeNSEServer struct{}
+type setregtimeNSEServer struct {
+	genericsync.Map[string, *timestamppb.Timestamp]
```
Comment (denis-tingaikin) [inline pkg/registry/common/setregistrationtime/nse_server.go:35]: Can we use metadata instead?

Comment (glazychev-art) [reply pkg/registry/common/setregistrationtime/nse_server.go:35]: @denis-tingaikin 
I thought about it, but unfortunately we don't have `metadata` for registry

Comment (denis-tingaikin) [reply pkg/registry/common/setregistrationtime/nse_server.go:35]: It might be a good time to add it ;)

Comment (glazychev-art) [reply pkg/registry/common/setregistrationtime/nse_server.go]: Fixed :)

**Comment by glazychev-art** [comment]
@denis-tingaikin 
Updated - https://github.com/networkservicemesh/sdk-k8s/pull/439

**Pull request merged**